scylla-cql/src/types/deserialize/value.rs

```diff
@@ -0,0 +1,1515 @@
+            &ColumnType::Timestamp,
+            serialize_cell(&Timestamp(timestamp1)),
+        );
+        compat_check::<Duration>(
+            &ColumnType::Timestamp,
+            serialize_cell(&Timestamp(timestamp2)),
+        );
+        compat_check::<Duration>(
+            &ColumnType::Timestamp,
+            serialize_cell(&Timestamp(timestamp3)),
+        );
+
+        compat_check::<DateTime<Utc>>(
+            &ColumnType::Timestamp,
+            serialize_cell(&Timestamp(timestamp1)),
+        );
+        compat_check::<DateTime<Utc>>(
+            &ColumnType::Timestamp,
+            serialize_cell(&Timestamp(timestamp2)),
+        );
+        compat_check::<DateTime<Utc>>(
+            &ColumnType::Timestamp,
+            serialize_cell(&Timestamp(timestamp3)),
+        );
+
+        // duration
+        let duration1 = CqlDuration {
+            days: 123,
+            months: 456,
+            nanoseconds: 789,
+        };
+        let duration2 = CqlDuration {
+            days: 987,
+            months: 654,
+            nanoseconds: 321,
+        };
+        compat_check::<CqlDuration>(&ColumnType::Duration, serialize_cell(&duration1));
+        compat_check::<CqlDuration>(&ColumnType::Duration, serialize_cell(&duration2));
+
+        // text types
+        for typ in &[ColumnType::Ascii, ColumnType::Text] {
+            compat_check::<String>(typ, make_cell("".as_bytes()));
+            compat_check::<String>(typ, make_cell("foo".as_bytes()));
+            compat_check::<String>(typ, make_cell("superfragilisticexpialidocious".as_bytes()));
+        }
+
+        // blob
+        compat_check::<Vec<u8>>(&ColumnType::Blob, make_cell(&[]));
+        compat_check::<Vec<u8>>(&ColumnType::Blob, make_cell(&[1, 9, 2, 8, 3, 7, 4, 6, 5]));
+
+        let ipv4 = IpAddr::from([127u8, 0, 0, 1]);
+        let ipv6: IpAddr = Ipv6Addr::LOCALHOST.into();
+        compat_check::<IpAddr>(&ColumnType::Inet, make_ip_address(ipv4));
+        compat_check::<IpAddr>(&ColumnType::Inet, make_ip_address(ipv6));
+
+        // uuid and timeuuid
+        // new_v4 generates random UUIDs, so these are different cases
+        let uuid1 = Uuid::new_v4();
+        let uuid2 = Uuid::new_v4();
+        let uuid3 = Uuid::new_v4();
+        compat_check::<Uuid>(&ColumnType::Uuid, serialize_cell(&uuid1));
+        compat_check::<Uuid>(&ColumnType::Uuid, serialize_cell(&uuid2));
+        compat_check::<Uuid>(&ColumnType::Uuid, serialize_cell(&uuid3));
+        compat_check::<Uuid>(&ColumnType::Timeuuid, serialize_cell(&uuid1));
+        compat_check::<Uuid>(&ColumnType::Timeuuid, serialize_cell(&uuid2));
+        compat_check::<Uuid>(&ColumnType::Timeuuid, serialize_cell(&uuid3));
+
+        // nulls, represented via option
+        compat_check::<Option<i32>>(&ColumnType::Int, serialize_cell(&123i32));
+        compat_check::<Option<i32>>(&ColumnType::Int, make_null());
+
+        // empty values
+        // TODO: FromCqlVal doesn't have support for this
```

Comment:
Please clarify: what must be done in order to test empty values here?

Reply:
Eh, they probably just need a separate test. I don't think we have an equivalent to the new `MaybeEmpty`, just the `CqlValue::Empty`, so `compat_check` can't really be used here.

We just need a test for `MaybeEmpty` - there isn't any at the moment. I'll add a test for it.

Reply:
Done